mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
hmm...actually the more I investigate, it seems that switching these lines around didn't actually result in the solution.  I might have just been lucky...in that the time needed to process the notes took less time simply by change.  That randomness leads me to to think that this might be a caching issue, or maybe a data structure like the r_b tree happened to be better organized at that time.